File: benchmarks/workloads/regex_match/python/bench.py
import re
import sys

n = int(sys.argv[1]) if len(sys.argv) > 1 else 3_000_000

email = re.compile(r"[a-z]+@[a-z]+")
number = re.compile(r"[0-9]+")
inputs = ["foo@bar", "12345", "hello world!"]

ce = cn = cx = 0
for i in range(n):
    s = inputs[i % 3]
    if email.fullmatch(s):
        ce += 1
    elif number.fullmatch(s):
        cn += 1
    else:
        cx += 1

print(f"email = {ce} number = {cn} none = {cx}")

File: benchmarks/workloads/regex_match/javascript/bench.js
const n = parseInt(process.argv[2] || "3000000", 10);

const email = /^[a-z]+@[a-z]+$/;
const number = /^[0-9]+$/;
const inputs = ["foo@bar", "12345", "hello world!"];

let ce = 0, cn = 0, cx = 0;
for (let i = 0; i < n; i++) {
    const s = inputs[i % 3];
    if (email.test(s)) ce++;
    else if (number.test(s)) cn++;
    else cx++;
}
console.log(`email = ${ce} number = ${cn} none = ${cx}`);
